poc/prg.py

```diff
@@ -136,7 +147,15 @@ def __init__(self, seed, dst, binder):
         #
         # Implementation note: This step can be cached across PRG
         # evaluations with many different seeds.
-        shake = cSHAKE128.new(custom=dst)
+        shake = SHAKE128.new()
+        dst_length_l = (len(dst).bit_length() + 7)//8
+        dst_length = len(dst).to_bytes(dst_length_l, byteorder='little')
+        dst_length_l = self.__to_bytes__(dst_length_l)  # one byte value
+        pre_pad = self.__to_bytes__(1) + self.__to_bytes__(168)
+        post_pad_l = (168 - ((len(dst) + len(dst_length)+3) % 168)) % 168
+        post_pad = (0).to_bytes(post_pad_l, byteorder='little')
+        dom_sep = pre_pad + dst_length_l + dst_length + dst + post_pad
```

Comment:
Can we deduplicate this logic, i.e., create a wrapper function for it and call it here and above?